### PR TITLE
CMA: make CMA work with Rendezvous.

### PR DESCRIPTION
**Pull request by alinask**

### CODE REVIEW

**Comment by alinask** [comment]
Related to https://github.com/openucx/ucx/issues/493

File: test/gtest/ucs/test_async.cc [hunk 1]
```diff
@@ -540,8 +540,7 @@ class local_event_add_handler : public local_event {
     local_event_add_handler(ucs_async_mode_t mode) :
         local_event(mode), m_event_set(false)
     {
-         int ret = pipe(m_pipefd);
-         ucs_assertv(0 == ret, "%m");
+         ucs_assertv(0 == pipe(m_pipefd), "%m");
```
Comment (yosefe) [inline test/gtest/ucs/test_async.cc]: don't see how this fix is related..
also the call to pipe() will be removed in release mode - and we don't want that

Comment (alinask) [inline test/gtest/ucs/test_async.cc]: when compiling ucx in release mode with gtest, 'ret' is defined but isn't used.

Comment (yosefe) [inline test/gtest/ucs/test_async.cc]: then need to use ucs_assert_always()
[without removing 'ret']

Comment (alinask) [inline test/gtest/ucs/test_async.cc]: ok, will update this.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/624/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3072/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/3073/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/625/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/2576/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
👍

**Pull request merged**